support downloading to a writer

mikewiacek opened the pull request:
Support downloading a zone file to an io.Writer allows code reuse and the hidden effects of calling os.Create within a function.

mikewiacek committed:
support downloading to a writer

lanrat commented on zones.go:
```diff
@@ -16,31 +16,30 @@ type DownloadInfo struct {
-	}
-	if n == 0 {
-		return fmt.Errorf("%s was empty", destinationPath)
-	}
```
Can you add this empty check into `DownloadZoneToWriter`?

mikewiacek replied:
I'd argue that an empty response body is not an error. If you look at io.Copy's semantics, or anything in the standard library, an error is more traditionally used to indicate some exceptional condition. io.Copy returns an int64 to inform the user of what was actually copied and doesn't overload an error for that case. I'll make a change that doesn't change the behavior of DownloadZone but makes DownloadZoneToWrite behave more idiomatically.

lanrat reviewed (changes requested):
Thanks for the pull request!

This looks good, but could you add the empty error check to `DownloadZoneToWriter`?

mikewiacek committed:
Update DownloadZoneToWriter 

Return the number of bytes written to an io.Writer
Update DownloadZone to return an error if no data was written to a file.

mikewiacek commented:
This should do what you'd like and make the overall behavior more idiomatic.

lanrat commented on zones.go:
```diff
@@ -16,23 +16,30 @@ type DownloadInfo struct {
+func (c *Client) DownloadZoneToWriter(url string, dest io.Writer) (int64, error) {
 	resp, err := c.apiRequest(true, "GET", url, nil)
 	if err != nil {
 		return err
```
You need to return an int64 and error here.

mikewiacek replied:
Oops, that's what I get for editing code from my phone. ;-) Sorry!

lanrat reviewed (changes requested):
Thanks for the update, however it looks like you forgot to add the new int64 in the return preventing this from compiling.

mikewiacek committed:
Fix a missing return argument.

lanrat commented:
LGTM!
